cmd/go-yaml/node.go

```diff
@@ -87,24 +135,199 @@ func formatStyle(s yaml.Style) string {
 		return "Folded"
 	case yaml.FlowStyle:
 		return "Flow"
+	case 0:
+		// Plain style - only show if profuse
+		if profuse {
+			return "Plain"
+		}
 	}
 	return ""
 }
 
+// formatStyleName converts a YAML node style into a lowercase style name.
+// Always returns a style name (defaults to "plain" for style 0).
+func formatStyleName(s yaml.Style) string {
+	// Remove tagged style bit for checking base style
+	baseStyle := s &^ yaml.TaggedStyle
+
+	switch baseStyle {
+	case yaml.DoubleQuotedStyle:
+		return "double"
+	case yaml.SingleQuotedStyle:
+		return "single"
+	case yaml.LiteralStyle:
+		return "literal"
+	case yaml.FoldedStyle:
+		return "folded"
+	case yaml.FlowStyle:
+		return "flow"
+	default:
+		return "plain"
+	}
+}
+
 // formatTag converts a YAML tag string to its string representation.
-func formatTag(tag string, style yaml.Style) string {
+func formatTag(tag string, style yaml.Style, profuse bool) string {
```

Comment:
Nitpicking:

I'm not fan of the boolean flag here.

It leads to have cryptic/magical boolean flag when calling the method.

I prefer passing one struct as argument with 3 fields instead of having 3 arguments